stock_change_qty_reason/views/stock_inventory_line_view.xml

```diff
@@ -19,8 +19,14 @@
             eval="[(4, ref('stock_change_qty_reason.group_qty_reason_preset'))]"
         />
         <field name="arch" type="xml">
+            <xpath expr="//field[@name='product_id']" position="before">
+                <field name="company_id" invisible="1" />
+            </xpath>
             <field name="reason" position="replace">
-                <field name="preset_reason_id" />
+                <field
+                    name="preset_reason_id"
+                    domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"
```

Comment:
Instead of doing this, I think it is better if you add multi company security rules for `stock.inventory.line.reason`

Reply:
correct me if I am wrong. In V13 you can have multiple companies active at the same time. so reason can not be from the current company of user but it has to be the company from the stock scrap/Stock picking is in

Reply:
I see your point now, thanks for sharing. In any case, adding default security rules is still a good practice when you add the company_id field to some model so it can coexist with this view teak, would you mind adding them?

Reply:
I agree with you adding additional security rule based on allowed company ids will be a good thing.

Reply:
Done